src/changeKeys.js

```diff
@@ -4,18 +4,29 @@ import pressedAnotherOperator from "./pressedAnotherOperator";
 const changeKeys = (result, button) => {
   const lastCharIsNum = "0123456789".includes(result[result.length - 1]);
   const lastCharIsOperator = "/*+-".includes(result[result.length - 1]);
+  const isOperator = "/*+-".includes(button);
 
   if (result === "") {
-    if ("+-(0123456789".includes(button)) {
+    if ("(0123456789".includes(button)) {
       return button;
     }
+    if (isOperator) {
+      return (result = "0" + button);
```

Comment:
why are you assigning result here?

Reply:
It doesn't need to be assigned, it can be return `"0" + button`, I played around with returning in each if block, and at the end so I had removed and replaced this code a lot but changeKeys is no longer being referenced anywhere in the latest push since I replaced its functionality in updateDisplay.js

Reply:
There are several places where you combine setting variables and returning the result of setting variables. I want to know what inspires you to do that and if you know what's happening when you do?

Reply:
_where you combine setting variables_

Do you mean like in `updateDisplay.js` or `changeKeys`, by setting the value of result in each if block and then returning at the end? I want to make sure I am understanding your question, sorry if this is tedious.

Reply:
If that is what you're referring to ^ I am thinking that it is returning result and button combined as one value that is `nextResult` in 
 ```
   if (button.isDisplayable) {
      nextResult = updateDisplay(result, button, lastEquation);

      this.setState({
        result: nextResult + button,
      });
    }
```

Which I thought would work because before introducing this way of setting these variables the tests were passing with:
```
    if (numOrDecimal || openParens || closeParens) {
      this.setState({
        result: result + button.key,
      });
    } else if (operatorOrDecimal) {
      this.setState({
        result: changeKeys(result, button.key),
      });
    }
```
So that result could be set as nextResult replacing the else if block, and button being added to result in the setState, replacing the first if block.

Reply:
`updateDisplay` doesn't have what I'm referring to. It doesn't assign variable values and return in the same statement or _on the same line_.

Reply:
Ah okay, I think you mean like here:
`  if (result === "") {
    if (isOperator) {
      return (result = "0" + button);
    }
  }`

Reply:
> Ah okay, I think you mean like here:
> 
> `  if (result === "") {
> 
>     if (isOperator) {
> 
>       return (result = "0" + button);
> 
>     }
> 
>   }`
> 
> 
> 
> What's happening there is that it is returning that result (combined with the button as one), and exiting out of that if block.

Exactly

Reply:
Am I correct there on what it is returning?

Reply:
You are correct! Just wanted to make sure.